### DISCUSSION TITLE
PTMC interaction energies calculation

### DISCUSSION THREAD

**Discussion by mugikumar** [ORTools4MTEX/ORTools#30] [Q&A] [answered]
Hi Frank,

I have used your martensite calculator code to predict which variants come first. I am getting the error when using the code. please look into the file attached to get the solution.

```matlab
scrPrnt('Step',sprintf('Defining %.0f Stress tensors',length(Sig.labels(Sig.choice))));
[Sig.ten] = cmptSigma(Sig.labels(Sig.choice));                             %Get stress tensors
% Compute Orientation distribution section
scrPrnt('Step',sprintf('Defining %s OD section at %.0f°',OD.info{1},OD.info{2}));
OD = cmptOD(CS.A,OD,SS);                                                   %Compute parent OD
% *******************   Initialize deformation mode
scrPrnt('SegmentStart','Initializing deformation mode');
% Calculating PTMC solution
scrPrnt('Step','Calculating PTMC solution');   
PTMC = mainPTMC(CS.A,CS.B,Def.SPa,Def.SDa,Def.bCa);                        %Compute PTMC solution
OR = cmptOR(CS,Def,PTMC);                                                  %Compute orientation relationship and variants
PTMC = analyzePTMC(PTMC,Def,CS,OR,Post,'symmetrise'); 
```

![martensite calculator error](https://user-images.githubusercontent.com/109027461/216016793-1c08f557-1912-4933-8e64-67046ac6c139.png)

**Comment by frankNiessen** [comment]
Hi Mugikumar, 

thank you for pointing this out, this should be resolved in an updated version I just released: [click here](https://github.com/frankNiessen/martensiteWorkCalculator/releases/tag/v1.0.1)

Best wishes
Frank

**Comment by mugikumar** [comment]
Hi Frank,
       Thanks for your reply, still i am getting the same error.

**Reply by JDFIJ** [reply]
I'm experiencing the same error problem as well, and I'm not sure if there is a solution.

**Reply by frankNiessen** [reply, marked as answer]
Hi, 

I apologize for the long wait, things have been a bit busy on my end. 
Could you try this version and let me know whether the error is resolved? It works on my system. 
[martensiteWorkCalculator v1.0.2](https://github.com/frankNiessen/martensiteWorkCalculator/releases/tag/v1.0.2)

If you get an error again, could you confirm whether it is the same error as stated above or a different type of error? Thanks

Best wishes
Frank

**Reply by JDFIJ** [reply]
Thank you very much. The program runs smoothly and calculates the results. I'd like to ask, how can I modify the program to obtain different variants of shapes and strains?or total shape strain in variant combinations in a packet?. Is there any method to do this?
![图片1](https://github.com/ORTools4MTEX/ORTools/assets/111446806/9e94ce1b-476c-4b27-af00-8d9ebd89e296)
https://www.sciencedirect.com/science/article/pii/S1359645413003509?via%3Dihub#b0035

**Comment by JDFIJ** [comment]
I'm experiencing the same error problem as well, and I'm not sure if there is a solution.
